### PR TITLE
fix: ensure mas builds of the same application can use safestorage

### PR DESCRIPTION
**Pull request by MarshallOfSound**
This change ensures that MAS builds of applications with an equivalent darwin build that share the same name do not fight over access to the same Safe Storage account.

Specifically this changes the account name for app "My App" from "My App" to "My App AppStore" if the app is using a MAS build of Electron.

We attempt to migrate the safe storage key from the old account, if that migration succeeds we delete the old key and move on.

Existing apps that aren't built for the app store should be unimpacted, there is one edge case where a user uses BOTH an AppStore and a darwin build of the same app only one will keep it's access to the SafeStorage key as during the migration we delete the old account.  This is an acceptable edge case as no one should be actively using two versions of the same app and I don't think should be classified as a breaking change because this scenario didn't work before anyway (due to conflicting IAM access to the same keychain item)

Notes: The keychain account name for MAS and Darwin builds of the same Electron application now differ

### CODE REVIEW

File: patches/chromium/feat_ensure_mas_builds_of_the_same_application_can_use_safestorage.patch [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
++    if (error == noErr) {
++      // If we successfully made the suffixed account we can delete the old
++      // account to ensure new apps don't try to use it and run into IAM
++      // issues
++      error = keychain_.ItemDelete(item_ref.get());
++      if (error != noErr) {
++        OSSTATUS_DLOG(ERROR, error) << "Keychain delete for legacy key failed";
++      }
++    } else {
++      OSSTATUS_DLOG(ERROR, error) << "Keychain add for suffixed key failed";
```
Comment (deepak1556) [inline patches/chromium/feat_ensure_mas_builds_of_the_same_application_can_use_safestorage.patch:90]: Should this error case return `std::string()` instead ?

Comment (MarshallOfSound) [reply patches/chromium/feat_ensure_mas_builds_of_the_same_application_can_use_safestorage.patch:90]: I don't think so, in this case we read the valid key but failed to migrate it.  Given we have a valid key we may as well use it.

I believe hitting this exact conditional is impossible without a corrupt keychain fwiw

File: patches/chromium/feat_ensure_mas_builds_of_the_same_application_can_use_safestorage.patch [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
++        service_name.size(), service_name.data(), suffixed_account_name.size(),
++        suffixed_account_name.data(), password.size(), password_data, NULL);
++
++    if (error == noErr) {
++      // If we successfully made the suffixed account we can delete the old
++      // account to ensure new apps don't try to use it and run into IAM
++      // issues
++      error = keychain_.ItemDelete(item_ref.get());
++      if (error != noErr) {
++        OSSTATUS_DLOG(ERROR, error) << "Keychain delete for legacy key failed";
```
Comment (deepak1556) [inline patches/chromium/feat_ensure_mas_builds_of_the_same_application_can_use_safestorage.patch:87]: Not sure when this case happens, but if we fail to delete the old account and successfully copy over the new account, seems like successive access would no longer hit this path to delete the old account. Do we leave the old account around in that case ?

**Review by deepak1556 — APPROVED** [review]
LGTM

File: patches/chromium/feat_ensure_mas_builds_of_the_same_application_can_use_safestorage.patch (hunk 2 above)
Comment (MarshallOfSound) [reply patches/chromium/feat_ensure_mas_builds_of_the_same_application_can_use_safestorage.patch:87]: Yeah, I think that's ok.  The alternative is we try to delete it on successful reads of the suffixed key.  But deleting it requires a lookup and that could throw up a unnecessary permission prompt.  I'd rather just leave the old key there.  I don't think a delete failure is likely based on my understanding of the keychain

**Review by VerteDinde — APPROVED** [review]

**Pull request merged**